<<<<<<< HEAD:package/python3/0015-Add-an-option-to-disable-the-curses-module.patch
From 962deb43f5f340a8472d432ae8e6d61186f9040e Mon Sep 17 00:00:00 2001
=======
From 32ecd62c7f1444652911d05c2c03152ff7fff6b5 Mon Sep 17 00:00:00 2001
>>>>>>> upstream-2025.02.7:package/python3/0009-Add-an-option-to-disable-the-curses-module.patch
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 22 Feb 2017 17:31:51 -0800
Subject: [PATCH] Add an option to disable the curses module

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[ Andrey Smirnov: ported to Python 3.6 ]
Signed-off-by: Andrey Smirnov <andrew.smirnov@gmail.com>
[ Adam Duskett: ported to Python 3.10.0 ]
Signed-off-by: Adam Duskett <aduskett@gmail.com>
---
<<<<<<< HEAD:package/python3/0015-Add-an-option-to-disable-the-curses-module.patch
 Makefile.pre.in |  4 +++-
 configure.ac    | 9 +++++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index c968113..f89f155 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1432,7 +1432,6 @@ LIBSUBDIRS=	asyncio \
=======
 Makefile.pre.in | 5 ++++-
 configure.ac    | 7 +++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 5de6f63220d..2b0307b1282 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -2095,7 +2095,6 @@ LIBSUBDIRS=	asyncio \
>>>>>>> upstream-2025.02.7:package/python3/0009-Add-an-option-to-disable-the-curses-module.patch
 		concurrent concurrent/futures \
 		csv \
 		ctypes ctypes/macholib \
-		curses \
 		dbm \
 		email email/mime \
<<<<<<< HEAD:package/python3/0015-Add-an-option-to-disable-the-curses-module.patch
@@ -1546,6 +1545,9 @@ TESTSUBDIRS += tkinter/test tkinter/test/test_tkinter \
 	tkinter/test/test_ttk
=======
 		encodings \
@@ -2261,6 +2260,10 @@ LIBSUBDIRS += tkinter
 TESTSUBDIRS += test/test_tkinter test/test_ttk
>>>>>>> upstream-2025.02.7:package/python3/0009-Add-an-option-to-disable-the-curses-module.patch
 endif
 
+ifeq (@CURSES@,yes)
+LIBSUBDIRS += curses
+endif
+
 COMPILEALL_OPTS=-j0
 
<<<<<<< HEAD:package/python3/0015-Add-an-option-to-disable-the-curses-module.patch
 TEST_MODULES=@TEST_MODULES@
 libinstall:	build_all $(srcdir)/Modules/xxmodule.c
diff --git a/configure.ac b/configure.ac
index dc76dff..4f1cda5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3391,6 +3391,15 @@ if test "$TK" = "no"; then
    DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _tkinter"
 fi
=======
 ifeq (@PYDOC@,yes)
diff --git a/configure.ac b/configure.ac
index 4331ed4264f..94bebbfd48e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4526,6 +4526,13 @@ AC_ARG_ENABLE(sqlite3,
 AS_IF([test "$SQLITE3" = "no"],
       [PY_STDLIB_MOD_SET_NA([_sqlite3])])
>>>>>>> upstream-2025.02.7:package/python3/0009-Add-an-option-to-disable-the-curses-module.patch
 
+AC_SUBST(CURSES)
+AC_ARG_ENABLE(curses,
+	AS_HELP_STRING([--disable-curses], [disable curses]),
+	[ CURSES="${enableval}" ], [ CURSES=yes ])
+AS_IF([test "$CURSES" = "no"],
+      [PY_STDLIB_MOD_SET_NA([_curses], [_curses_panel])])
+
 AC_SUBST(PYDOC)
 
 AC_ARG_ENABLE(pydoc,
-- 
<<<<<<< HEAD:package/python3/0015-Add-an-option-to-disable-the-curses-module.patch
2.30.2
=======
2.39.5
>>>>>>> upstream-2025.02.7:package/python3/0009-Add-an-option-to-disable-the-curses-module.patch

